File: binaryheap.java
public class BinaryHeapExample {
    public static void main(String[] args) {
        // Бинарная куча через PriorityQueue
        PriorityQueue<Integer> heap = new PriorityQueue<>();

        heap.add(50);
        heap.add(20);
        heap.add(70);
        heap.add(10);
        heap.add(90);

        // Вывести и удалить элементы в куче в порядке возрастания
        System.out.println("Элементы в порядке возрастания:");
        while (!heap.isEmpty()) {
            System.out.print(heap.poll() + " ");
        }
        System.out.println();
    }
}

File: binaryheap.cpp
#include <iostream>
#include <queue>

int main() {
    // Бинарная куча через priority_queue
    std::priority_queue<int> heap;

    heap.push(50);
    heap.push(20);
    heap.push(70);
    heap.push(10);
    heap.push(90);

    // Вывести элементы по порядку
    std::cout << "Элементы в куче:" << std::endl;
    while (!heap.empty()) {
        std::cout << heap.top() << " "; // Вывести максимальный элемент
        heap.pop();                     // Убрать его
    }
    std::cout << std::endl;

    return 0;
}
